azurelinuxagent/common/utils/shellutil.py

```diff
@@ -117,30 +117,51 @@ def _encode_command_output(output):
     return ustr(output, encoding='utf-8', errors="backslashreplace")
 
 
-def run_command(command):
+class CommandError(Exception):
```

Comment:
I know that is what we've doing so far. but I do not think that is a good pattern. Basically we are dumping small pieces (the exceptions) from completely separate modules into a single file. I believe it is cleaner to define the exception in the module that originates it, to avoid unneeded/unwanted cross-module dependencies.